tools/openapi-plugin/src/generators/client/generator.ts

```diff
@@ -113,4 +122,27 @@ async function copySchema(
     }
 }
 
+async function validate(schemaPath: string) {
+    return new Promise((resolve, reject) => {
+        const child = spawn('npx', [
+            'nx',
+            'run',
+            '@aligent/openapi-plugin:validate',
+            schemaPath,
+        ]);
```

Comment:
Why don't we just directly call `npx @redocly/cli lint` here? Besides, does the @redocly expose api for us instead of spawning a child process?

Reply:
So the exposed API does exist, but it comes with a few problems. The most prevalent being that the `redocly/openapi-core` package is almost completely undocumented and designed for internal use (which the developers themselves specify). 

The second problem is that using the API doesn't automatically produce an output on the CLI. The output of the APIs lint (which is undocumented) would have to be processed and transformed, to produce some kind of readable output for the user of the plugin. This is doable, but in my opinion it seems like a much easier option to just spawn a child process and run the cli version instead, which already produces a nicely formatted output to the console, without us having to do any of the work. 

The direct call to the linter is a good idea though. Will change :+1: